src/jira.ts

```diff
@@ -79,8 +79,10 @@ export class Jira {
         estimate: typeof estimate === 'string' || typeof estimate === 'number' ? estimate : 'N/A',
         labels,
       };
-    } catch (e) {
-      throw e;
+    } catch (error) {
+      console.log(`The JIRA issue key ${key} is not valid`);
+      console.log({ error });
+      process.exit(1);
```

Comment:
this will cause the process to halt at the first instance that a JIRA key was not found. However - the regex provides us multiple matches - we should be testing with all of those. And throwing an error if none of those match.

a test case for this would be a branch named `TEST-123-do-something-CLIN-1857` where TEST-123 is an invalid JIRA and CLIN-1857 is valid.

we should also add test cases within this repository for these scenarios

Reply:
We want to halt the process for the current scenario as it is not running in the loop. If you look at the [code](https://github.com/adityasingh02/action-jira-linter/blob/main/src/main.ts#:~:text=const%20issueKey%20%3D,issueKey)%3B), we are using the last issue key to fetch the JIRA id. So if the same issue_id is not found in the JIRA then halt the process. 

The scenario you are mentioning is a part of my current [ticket](https://fireflyhealth.atlassian.net/browse/CLIN-1588) and I will use the example you mentioned. But for this PR, it is just for a single issue key. (I will also remove this halt in the upcoming PR so it won't exit from a single process).

Reply:
Regarding the test case, I was thinking of writing but thought I will be removing the process exit code in the upcoming PR, so it won't make sense just to add it for a temporary purpose. But if you feel the need then I can add.

Reply:
if we do want to exit - the exit code should be in the `main.ts` file and reuse the existing `exit` functionality which would honor the `failOnError` flag and mark the action as failed.

Reply:
Added in the PR https://github.com/adityasingh02/action-jira-linter/pull/4